service/task.go

```diff
@@ -840,67 +791,238 @@ func (uis *UIServer) taskModify(w http.ResponseWriter, r *http.Request) {
+				return
+			}
+		}
+
+		if testLog == nil {
+			http.Error(w, "not found", http.StatusNotFound)
 			return
 		}
+
+		ctx, cancel := context.WithCancel(r.Context())
+		defer cancel()
+
+		data.Data = make(chan apimodels.LogMessage)
+		go func() {
+			defer close(data.Data)
+			for _, line := range testLog.Lines {
+				if ctx.Err() != nil {
+					return
+				}
+				data.Data <- apimodels.LogMessage{
+					Type:     apimodels.TaskLogPrefix,
+					Severity: apimodels.LogInfoPrefix,
+					Version:  evergreen.LogmessageCurrentVersion,
+					Message:  line,
+				}
+			}
+		}()
+	}
+
+	template := "task_log.html"
+	if raw {
+		template = "task_log_raw.html"
+		uis.renderText.Stream(w, http.StatusOK, data, "base", template)
 	} else {
-		taskID := vars["task_id"]
-		testName := vars["test_name"]
-		taskExecutionsAsString := vars["task_execution"]
-		taskExec, err = strconv.Atoi(taskExecutionsAsString)
-		if err != nil {
-			http.Error(w, "task execution num must be an int", http.StatusBadRequest)
-			return
+		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
+	}
+}
+
+func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, projCtx projectContext, uiTask *uiTaskData, execution int) {
+	ctx := r.Context()
+	var err error
+
+	uiTask.TestResults = []uiTestResult{}
+	execTasks := []task.Task{}
+	execTaskIDs := []string{}
+	if uiTask.DisplayOnly {
+		for _, t := range projCtx.Task.ExecutionTasks {
+			var et *task.Task
+			if uiTask.Archived {
+				et, err = task.FindOneOldNoMergeByIdAndExecution(t, execution)
+			} else {
+				et, err = task.FindOneId(t)
+			}
+			if err != nil {
+				uis.LoggedError(w, r, http.StatusInternalServerError, err)
+				return
+			}
+			if et == nil {
+				grip.Error(message.Fields{
+					"message": "execution task not found",
+					"task":    t,
+					"parent":  projCtx.Task.Id,
+				})
+				continue
+			}
+			execTasks = append(execTasks, *et)
+			execTaskIDs = append(execTaskIDs, t)
 		}
 
-		testLog, err = model.FindOneTestLog(testName, taskID, taskExec)
+		execTasks, err = task.MergeTestResultsBulk(execTasks, nil)
 		if err != nil {
 			uis.LoggedError(w, r, http.StatusInternalServerError, err)
 			return
 		}
+		for i, execTask := range execTasks {
+			uiTask.ExecutionTasks = append(uiTask.ExecutionTasks, uiExecTask{
+				Id:        execTaskIDs[i],
+				Name:      execTask.DisplayName,
+				TimeTaken: execTask.TimeTaken,
+				Status:    execTask.ResultStatus(),
+			})
+		}
 	}
 
-	if testLog == nil {
-		http.Error(w, "not found", http.StatusNotFound)
-		return
-	}
+	// Check cedar test results first.
+	success := true
+	if uiTask.DisplayOnly {
+		allResults := []uiTestResult{}
+		for i, execTask := range execTasks {
+			opts := apimodels.GetCedarTestResultsOptions{
+				BaseURL:   uis.Settings.Cedar.BaseURL,
+				TaskID:    execTask.Id,
+				Execution: execTask.Execution,
+			}
+			testResults, err := apimodels.GetCedarTestResults(ctx, opts)
```

Comment:
This would be one request per execution task, cedar does not have a concept of display tasks. Each request gets all tests for a single task.

Reply:
Oops I misread that part, but the danger is still there. Even doing 1 request per execution task means opening some pages will hammer cedar's network. Resmoke currently has a max of 1000 execution tasks it would generate, and changing a lot of files in the server codebase would cause it to hit that max

Reply:
Sounds like we should make a new rest route in cedar, I will make a follow up ticket.

Reply:
https://jira.mongodb.org/browse/EVG-13419